cpp/include/cudf/detail/aggregation/aggregation.cuh

```diff
@@ -228,10 +228,10 @@ struct update_target_element<
 
     using Target = target_type_t<Source, aggregation::ARGMAX>;
     auto old     = atomicCAS(&target.element<Target>(target_index), ARGMAX_SENTINEL, source_index);
-    if (old == ARGMAX_SENTINEL) { return; }
-
-    while (source.element<Source>(source_index) > source.element<Source>(old)) {
-      old = atomicCAS(&target.element<Target>(target_index), old, source_index);
+    if (old != ARGMAX_SENTINEL) {
+      while (source.element<Source>(source_index) > source.element<Source>(old)) {
+        old = atomicCAS(&target.element<Target>(target_index), old, source_index);
+      }
```

Comment:
Changes to this file seem unnecessary. The logic is the same. Why was this changed?

Reply:
To execute this line
https://github.com/karthikeyann/cudf/blob/625f52c3f8a4e94a2883dd182597445a05c7fcc4/cpp/include/cudf/detail/aggregation/aggregation.cuh#L237
old `return` statement skips setting valid bit.
This change sets valid bit.